compiler/src/dmd/dsymbolsem.d

```diff
@@ -5892,6 +5892,31 @@ void addEnumMembers(EnumDeclaration ed, Scope* sc, ScopeDsymbol sds)
     });
 }
 
+/******************************************************
+ * Verifies if the given Identifier is a DRuntime hook. It uses the hooks
+ * defined in `id.d`.
+ *
+ * Params:
+ *  id = Identifier to verify
+ * Returns:
+ *  true if `id` is a DRuntime hook
+ *  false otherwise
+ */
+private bool isDRuntimeHook(Identifier id)
+{
+    return id == Id._d_HookTraceImpl ||
+        id == Id._d_newclassT || id == Id._d_newclassTTrace ||
+        id == Id._d_arraycatnTX || id == Id._d_arraycatnTXTrace ||
+        id == Id._d_newThrowable || id == Id._d_delThrowable ||
+        id == Id._d_arrayassign_l || id == Id._d_arrayassign_r ||
+        id == Id._d_arraysetassign || id == Id._d_arraysetctor ||
+        id == Id._d_arrayctor ||
+        id == Id._d_arraysetlengthTImpl || id == Id._d_arraysetlengthT ||
+        id == Id._d_arraysetlengthTTrace ||
+        id == Id._d_arrayappendT || id == Id._d_arrayappendTTrace ||
+        id == Id._d_arrayappendcTXImpl;
```

Comment:
I suggest getting the list that the backend currently uses. It's kind of problematic that we need to update this every time a new hook is templated.

Reply:
I thought about this, but AFAICT, the glue layer uses this `enum` [1], while the identifiers are strings. The glue layer uses `getSymbol()` [2] to convert `RTLSYM`s to `Symbol`s, which contain the names of the hooks. 

[1] https://github.com/dlang/dmd/blob/36043ded818506936790be55f9353d6132527541/compiler/src/dmd/backend/rtlsym.d#L19
[2] https://github.com/dlang/dmd/blob/36043ded818506936790be55f9353d6132527541/compiler/src/dmd/backend/drtlsym.d#L54